app/src/main/java/org/kiwix/kiwixmobile/di/components/ServiceComponent.kt

```diff
@@ -35,6 +36,8 @@ interface ServiceComponent {
 
     @BindsInstance fun service(service: Service): Builder
 
+    @BindsInstance fun notificationManager(notificationManager: NotificationManager): Builder
```

Comment:
you don't need to do this, in our applicationModule
```
  @Provides
  @Singleton
  internal fun provideNotificationManager(context: Context): NotificationManager =
    context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
```
We haven't got into the specifics of why DI Is especially complicated in a multimodule project but in CoreComponent we expose this dependency across modules like so
```
  fun notificationManager(): NotificationManager
```

Using `@BindsInstance` is saying "to create this component I _need_ these objects"